### ISSUE TITLE
Confusion about timestamp_t

### ISSUE THREAD

**Issue by lukas-krecan-s1** [ocsf/ocsf-schema#955]
The definition of `timestamp_t` is

> The nominal data type for these attributes is timestamp_t based on Unix time or number of milliseconds since the Unix epoch.

Unfortunately, there are multiple interpretations

1. People who check the definition of [Unix Time](https://en.wikipedia.org/wiki/Unix_time) read "Unix Time is currently defined as the number of non-leap seconds which have passed since 00:00:00 UTC on Thursday, 1 January 1970" so they interpret the time as **seconds or milliseconds since epoch**
2. Some people take the "or number of milliseconds since the Unix epoch" as clarification of the Unix Time part so they interpret is as **milliseconds** since epoch
3. Some people read "..values are sometimes stored with higher granularity, such as microseconds or nanoseconds" and interpret it as **seconds, milliseconds, microseconds or nanoseconds**
Can you please reformulate it so it's clear what is meant. (Preferably just milliseconds should be allowed.)

**Comment by mmaitre314** [comment]
Adding to that request -- this is an area where being prescriptive about storage-format-specific features would help. For instance, I was trying to process the sample OCSF Parquet files at https://github.com/aws-samples/amazon-security-lake-ocsf-validation and timestamps came out as unit-less int64 like 1649721732000 when reading using Pandas Dataframe. Looking that up online, https://www.epochconverter.com/ says it's probably milliseconds but not sure. Requiring the use of Parquet's [TIMESTAMP](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#timestamp) would remove the need for that guesswork, along with associated bugs and incompatibilities. This would also allow data consumers to format the timestamps as they see fit (human readable, time zone, etc.).

**Comment by pagbabian-splunk** [comment]
I agree that we need to be more prescriptive.  The `time` topic was a big discussion early on, and external references need to be correctly adhered to.  We intended the `timestamp_t` to be milliseconds, and we needed the `datetime_t` type for readability options, but also for Parquet nanosecond possibilities as I recall.

Re: the profile, yes, the assumption that both should be produced is correct.  Needs to be made clear.

I will add this to this week's agenda for our weekly call (2/13/24).

**Comment by mavam** [comment]
The unit ambiguity discussed here is a symptom of a deeper issue: `timestamp_t` defines an encoding (epoch integer) where it should define semantics (a point in time). I filed #1694 proposing to fix it at the root, keeping the JSON wire format unchanged, so that columnar representations like Parquet/Iceberg can use native timestamp types conformantly (which also addresses the unit-less `int64` Parquet confusion @mmaitre314 reported above).
